bears/general/CasingBear.py

```diff
@@ -0,0 +1,137 @@
+import re
+
+from coalib.bears.LocalBear import LocalBear
+from coalib.bearlib.naming_conventions import (
+    to_camelcase, to_pascalcase, to_snakecase)
+from bears.general.AnnotationBear import AnnotationBear
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.results.SourceRange import SourceRange
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+
+
+class CasingBear(LocalBear):
+
+    def run(self,
+            filename,
+            file,
+            dependency_results,
+            casing: str,
+            language: str):
+        """
+        Checks whether all identifier names (variables, classes, objects)
+        follow a certain naming convention.
+
+        :param casing:   camelCasing or snake_casing or PascalCasing
+        :param language: The language of the file, which is used to determine
+                         the keywords to ignore.
+        """
+        casing_convention = {"camel": to_camelcase,
+                             "pascal": to_pascalcase,
+                             "snake": to_snakecase}
+
+        if casing not in casing_convention:
+            self.err("Invalid file-naming-convention provided: " + casing)
+            return
+
+        self.convertor = casing_convention[casing]
+        coalang = LanguageDefinition(language)
+
+        if "keywords" not in coalang or "special_chars" not in coalang:
```

Comment:
so this won't be required after https://github.com/coala-analyzer/coala/issues/2200 could you add a TODO so that we remember to remove it?